gnomad/utils/annotations.py

```diff
@@ -1140,65 +1168,68 @@ def hemi_expr(
-        == hl.locus(contig=chr_in, pos=int(pos_in), reference_genome=ref_input)
+    # build_in = gnomad.utils.filtering.get_reference_genome(ht.locus).name
+    ht = ht.filter(
+        (
+            ht.locus
+            == hl.locus(contig=chr_in, pos=int(pos_in), reference_genome=build_in)
+        )
+        & (ht.alleles == [ref_in, alt_in])
     )
-    table_filtered = table_filtered.filter(table_filtered.alleles == [ref_in, alt_in])
 
-    if table_filtered.count() != 1:
+    if ht.count() != 1:
         raise ValueError(
             "Error: can only work with one variant for this code, 0 or multiple"
-            " returned"
+            " returned."
         )
 
-    with hl.utils.hadoop_open(
-        "gs://gnomad-vrs-io-finals/chromosome_seqID_dictionary.json", "r"
-    ) as f:
-        chr_reread = json.load(f)
-
-    with hl.utils.hadoop_open(
-        "gs://gnomad-vrs-io-finals/example_schema.json", "r"
-    ) as f:
-        model_schema = json.load(f)
-
-    # update the model schema with: chromosome ID, sequence ID, start, end, and allele
-
-    variant_report = model_schema.replace("'", '"')  # formatting quirk to change ' to "
-    variant_report = json.loads(variant_report)  # loads it as a dictionary
+    variant_report = {
+        "_id": "Insert VRS_Allele_IDs",
+        "type": "Allele",
+        "location": {
+            "type": "SequenceLocation",
+            "sequence_id": "Insert ID from VRS_Chrom_IDs",
+            "interval": {
+                "type": "SequenceInterval",
+                "start": {
+                    "type": "Number",
+                    "value": "Insert Integer start location from VRS_Starts[1]",
+                },
+                "end": {
+                    "type": "Number",
+                    "value": "Insert Integer end location from VRS_Ends[1]",
+                },
+            },
+        },
+        "state": {
+            "type": "LiteralSequenceExpression",
+            "sequence": "Insert string state from VRS_States[1]",
+        },
+    }
 
-    variant_report["_id"] = table_filtered.info.vrs.VRS_Allele_IDs[1].collect()[0]
-    variant_report["location"]["sequence_id"] = chr_reread[chr_in]
-    variant_report["location"]["interval"]["start"] = str(
-        table_filtered.info.vrs.VRS_Starts[1].collect()[0]
+    variant_report["_id"] = ht.info.vrs.VRS_Allele_IDs[1].collect()[0]
+    variant_report["location"]["sequence_id"] = VRS_CHROM_IDS[chr_in]
+    variant_report["location"]["interval"]["start"]["value"] = str(
+        ht.info.vrs.VRS_Starts[1].collect()[0]
     )
-    variant_report["location"]["interval"]["end"] = str(
-        table_filtered.info.vrs.VRS_Ends[1].collect()[0]
+    variant_report["location"]["interval"]["end"]["value"] = str(
+        ht.info.vrs.VRS_Ends[1].collect()[0]
     )
-    variant_report["state"] = table_filtered.info.vrs.VRS_States[1].collect()[0]
+    variant_report["state"]["sequence"] = ht.info.vrs.VRS_States[1].collect()[0]
```

Comment:
u can drop all this code now and just build it into the above dictionary, such as:

variant_report = {"_id": f"{test.info.vrs.VRS_Allele_IDs[1].collect()[0]}",

Reply:
that's smart, but I'm a little worried about that dictionary itself looking pretty cluttered? I'll definitely give it a go, though!

Reply:
taking this approach moving forward, defining terms like this before the dict is defined, and just passing them in WHEN the dict is defined